metricbeat/schema/schema.go

```diff
@@ -71,19 +71,13 @@ func (o Object) HasKey(key string) bool {
 
 // ApplyTo adds the fields extracted from data, converted using the schema, to the
 // event map.
-func (s Schema) ApplyTo(event common.MapStr, data map[string]interface{}) (common.MapStr, *Errors) {
```

Comment:
Agree on the above. But that requires changes to the code as right now also missing optional fields return an error (and not nil)